### DISCUSSION TITLE
Xray vs Naive

### DISCUSSION THREAD

**Discussion by npwc** [XTLS/Xray-core#1993] [Q&A]
Hi 各位，有没有朋友对比了REALITY协议的Xray和Naiveproxy，性能和安全性如何？

**Comment by RPRX** [comment]
和 NaiveProxy 对标的应该是 XTLS Vision，它们都可以使用 REALITY，~~其实我早就想给 Naive 加了，最近有点拖延~~

REALITY 客户端非常简单、容易实现，服务端把 TLS 库换成 REALITY 库即可，~~所以等一个有缘人吧~~

**Reply by RPRX** [reply]
https://github.com/XTLS/naiveproxy-reality
https://github.com/XTLS/forwardproxy-reality

@H1JK 有没有兴趣

**Reply by H1JK** [reply]
Naive comes with multiplexing, so it should be more like h2/gRPC than XTLS(1:1)?

**Reply by RPRX** [reply]
> Naive comes with multiplexing, so it should be more like h2/gRPC than XTLS(1:1)?

其实主要是看中了它的真 Chrome 网络栈，当然，它也自带多路复用和 padding。
客户端给 BoringSSL 加上 REALITY 双向认证就行了，应该非常简单，甚至可以弄成实时打 patch 的形式。
服务端要改 Caddy、要加配置，有点麻烦，要不还是直接给 Xray 加 naive 版 forwardproxy 吧，~~虽然原汁原味的 Caddy 可能更好。~~

**Reply by ghost** [reply]
naive服务端的caddy用起来 我也觉得麻烦
